src/fs/file.rs

```diff
@@ -290,6 +291,56 @@ impl File {
         op.await
     }
 
+    /// Like [`read_at`], but using a pre-mapped buffer
+    /// registered with [`FixedBufRegistry`].
+    ///
+    /// [`read_at`]: Self::read_at
+    /// [`FixedBufRegistry`]: crate::buf::fixed::FixedBufRegistry
+    ///
+    /// # Errors
+    ///
+    /// In addition to errors that can be reported by `read_at`,
+    /// this operation fails if the buffer is not registered in the
+    /// current `tokio-uring` runtime.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    ///# fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// use tokio_uring::fs::File;
+    /// use tokio_uring::buf::fixed::FixedBufRegistry;
+    /// use tokio_uring::buf::IoBuf;
+    /// use std::iter;
+    ///
+    /// tokio_uring::start(async {
+    ///     let registry = FixedBufRegistry::new(iter::repeat(vec![0; 10]).take(10));
+    ///     registry.register()?;
+    ///
+    ///     let f = File::open("foo.txt").await?;
+    ///     let buffer = registry.check_out(2).unwrap();
+    ///
+    ///     // Read up to 10 bytes
+    ///     let (res, buffer) = f.read_fixed_at(buffer.slice(..), 0).await;
+    ///     let n = res?;
+    ///
+    ///     println!("The bytes: {:?}", &buffer[..n]);
+    ///
+    ///     // Close the file
+    ///     f.close().await?;
+    ///     Ok(())
+    /// })
+    ///# }
```

Comment:
Nice example. I think we should get something in, with the understanding that some things will change as we try to support Read and Write traits.